### PR TITLE
fix: enable bun.js by catching NotImplemented error

### PR DESCRIPTION
**Pull request by owlcode**
Catch [error thrown by bun.js](https://github.com/oven-sh/bun/blob/801e475c72b3573a91e0fb4c3afdd53b437770e1/src/js/node/perf_hooks.ts#L158) when calling `perf_hooks.monitorEventLoopDelay`

Following up on https://github.com/siimon/prom-client/pull/599 - bun.js already introduced some of the missing APIs (or created stubs) so to the best of my knowledge this is the last thing which is required to make default gc stats work for bun (at least it worked for me with this change).

Error is also reported in bun's repository - https://github.com/oven-sh/bun/issues/4708#issuecomment-2030436691

Fixes https://github.com/siimon/prom-client/issues/570
Closes https://github.com/siimon/prom-client/issues/623

### CODE REVIEW

File: lib/metrics/eventLoopLag.js [hunk 1]
```diff
@@ -37,32 +37,35 @@ module.exports = (registry, config = {}) => {
...
+				lagStddev.set(labels, histogram.stddev / 1e9);
+				lagP50.set(labels, histogram.percentile(50) / 1e9);
+				lagP90.set(labels, histogram.percentile(90) / 1e9);
+				lagP99.set(labels, histogram.percentile(99) / 1e9);
+
+				histogram.reset();
+			};
+		} catch {
+			// bun has `perf_hooks.monitorEventLoopDelay` defined but throws "NotImplemented" when called
+		}
```
Comment (SimenB) [inline lib/metrics/eventLoopLag.js:68]: the `catch` should check for the specific error bun throws and rethrow any other

File: lib/metrics/heapSpacesSizeAndUsed.js [hunk 2]
```diff
@@ -11,6 +11,11 @@ METRICS.forEach(metricType => {
 });
 
 module.exports = (registry, config = {}) => {
+	try {
+		v8.getHeapSpaceStatistics();
+	} catch {
+		return; // Bun
+	}
```
Comment (SimenB) [inline lib/metrics/heapSpacesSizeAndUsed.js:18]: ditto

**Review by SimenB — CHANGES_REQUESTED** [review]

**Comment by SimenB** [comment]
Can you include the CI stuff from #599?

**Comment by owlcode** [comment]
> Can you include the CI stuff from https://github.com/siimon/prom-client/pull/599?

Done

**Comment by SimenB** [comment]
https://github.com/siimon/prom-client/releases/tag/v15.1.2

**Pull request merged**